docs: configure s3 for file storage backend (#10678)
langflow-ai/langflow: docs/docs/Develop/concepts-file-management.mdx

Each Langflow server has a file management system where you can store files that you want to use in your flows.

Files uploaded to Langflow file management are stored in your configured storage backend and are available to all of your flows.
Langflow supports two storage backends: local filesystem storage and AWS S3 storage.

> Langflow uses local file storage as default. To configure Langflow to use S3 for file storage, see [configure file storage](/concepts-file-management#configure-file-storage).

Reviewer: ```suggestion
Files uploaded to Langflow file management are stored in Langflow's [storage backend (local or AWS S3)](/concepts-file-management#configure-file-storage), and they are available to all of your flows.
```

For videos, see the **Twelve Labs** and **YouTube** <Icon name="Blocks" aria-hidden="true" /> [**Bundles**](/components-bundle-components).

## Configure file storage

Langflow supports two storage backends for file management:

* **Local storage**: Langflow's default storage backend. Files are stored locally in your [Langflow configuration directory](/memory). Set `LANGFLOW_STORAGE_TYPE=local` or leave it unset to use local storage.

* **S3 storage**: Files are stored in an AWS S3 bucket.
To use S3 as your file storage backend, add the following configuration to your `.env` file:

```text
# S3 Storage Configuration
LANGFLOW_STORAGE_TYPE=s3
LANGFLOW_OBJECT_STORAGE_BUCKET_NAME=S3_BUCKET_NAME
LANGFLOW_OBJECT_STORAGE_PREFIX=S3_BUCKET_DIRECTORY

# AWS Credentials (required for S3)
AWS_ACCESS_KEY_ID=S3_ACCESS_KEY
AWS_SECRET_ACCESS_KEY=S3_ACCESS_SECRET_KEY
AWS_DEFAULT_REGION=S3_REGION
```

> Replace the following values:

Reviewer: ```suggestion
Replace the following placeholders with the actual values for your S3 instance:

```

* `S3_BUCKET_NAME`: The name of your S3 bucket.
* `S3_BUCKET_DIRECTORY`: An optional folder path within the bucket where files are stored, such as `s3://S3_BUCKET_NAME/S3_BUCKET_DIRECTORY`.
* `S3_ACCESS_KEY`: Your AWS Access Key ID.
* `S3_ACCESS_SECRET_KEY`: Your AWS Secret Access Key.
* `S3_REGION`: The AWS region where your bucket is located, such as `us-east-2`.

Your AWS credentials must have the necessary S3 permissions.

> Your requirements will vary depending on your use case.

Reviewer: ```suggestion
Your AWS credentials must have the necessary permissions to perform the required S3 operations for your use case, such as reading, writing, and deleting files in S3.
```

This example policy allows basic CRUD operations on S3 objects.

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "LangflowS3StorageAccess",
            "Effect": "Allow",
            "Action": [
                "s3:PutObject",
                "s3:GetObject",
                "s3:DeleteObject",
                "s3:ListBucket",
                "s3:PutObjectTagging",
            ],
            "Resource": [
                "arn:aws:s3:::S3_BUCKET_NAME",
                "arn:aws:s3:::S3_BUCKET_NAME/S3_BUCKET_DIRECTORY/*"
            ]
        }
    ]
}
```

> Replace the following values:

Reviewer: ```suggestion

Replace the following placeholders with the actual values for your IAM policy and S3 instance:

```
